### DISCUSSION TITLE
Getting Error: Uncaught (in promise): TypeError: Cannot read properties of null (reading 'getRequestInfo') at n.handleWindowCallback

### DISCUSSION THREAD

**Discussion by sonajoseph96** [cypress-io/cypress#26414] [Questions and Help]
When I try to Sign in to my application it gives error as below. Anyway to be able to fix this issue? 
1) Launch website
2) Click on Sign In
3) Enter username and click on Next
4) Enter password and click on Sign In
5) Click on Yes button on popup asking to remain signed in /// at this stage the error is thrown instead of taking user to the home page of the application

**CODE**
describe('template spec', () => {

  it('jjsLogin', () => {
    cy.visit('https://videoportaluat.azurewebsites.net/home',)

    cy.contains('Sign In').click()   //Sign In button
    
    cy.origin('https://login.microsoftonline.com', () => {
      cy.get('#i0116')
      .invoke('val', 'myusername')   //enter actual username  xxx@jjswaste.com.au
      .trigger('input')

     
      cy.get('#idSIButton9').click()    //Next button
    })
    
    cy.origin('https://login.microsoftonline.com', () => {
      cy.get('#i0118')
      .invoke('val', 'mypassword')   //enter actual password
      .trigger('input')

    
      cy.get('#idSIButton9').click()   // Sign in button

      cy.wait(1000)npx
      cy.get('#idSIButton9').click()  // Yes button
      

    })
  })
})

ERROR 
Unhandled Exception: Error: Uncaught (in promise): TypeError: Cannot read properties of null (reading 'getRequestInfo') TypeError: Cannot read properties of null (reading 'getRequestInfo') at n.handleWindowCallback (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:1404594) at e.handleCallback (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:826004) at e.canActivate (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:546320) at t.project (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:385923) at t._next (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:1377291) at t.next (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:89143) at t._subscribe (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:275508) at e._trySubscribe (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:66374) at e.subscribe (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:66203) at e.call (https://videoportaluat.azurewebsites.net/main.7f65065004dccc833631.bundle.js:1:1377064)

**Comment by portyu9** [comment]
Hi @sonajoseph96 

The stack trace indicates that this exception is being thrown by the **application's authentication callback code after the redirect**, not by `cy.origin()` itself:

```text
Cannot read properties of null (reading 'getRequestInfo')
...
handleWindowCallback
```

Using `cy.origin()` for the Microsoft login origin is still the supported Cypress approach. Cypress's current Azure AD guidance uses `cy.origin()` together with `cy.session()`.

I would first change the form interaction from manually setting the input value:

```js
.invoke('val', ...)
.trigger('input')
```

to normal Cypress typing:

```js
cy.get('input[type="email"]').type(username, { log: false })
cy.get('input[type="submit"]').click()

cy.get('input[type="password"]').type(password, { log: false })
cy.get('input[type="submit"]').click()
```

This makes the Microsoft form receive normal keyboard/input events.

However, the important problem is the exception after Azure redirects back to your application. The `handleWindowCallback` stack points to the application's authentication callback handling. The application needs to have the expected authentication request/state available when processing that callback.

Compare a successful manual login with the Cypress login and inspect:

- the redirect URI when Azure returns to the application;
- the callback URL/hash/query parameters;
- the authentication state stored by the application;
- any auth-library logging around `handleWindowCallback()`.

If the application still uses legacy ADAL-style callback handling, that is the area to investigate rather than suppressing Cypress's exception.

Do **not** fix this by returning `false` from `uncaught:exception`; that would only hide the application's failed login callback.

Once login is working, wrapping it in `cy.session()` is the current recommended way to avoid repeating the full Azure login for every test.

Please mark this answer as accepted if it helped, thank you.
